README.txt
$Id: README.txt,v 1.16 2011/01/09 18:31:52 dww Exp $

The signup module allows users to sign up for nodes of any type.  For
each signup-enabled node, this module provides options for sending a
notification email to a selected email address upon a new user signup
(good for notifying event coordinators, etc.), and a confirmation
email to users who sign up.  When used on event-enabled nodes
(http://drupal.org/project/event) or nodes that have a CCK date field
(http://drupal.org/project/date), it can also send out reminder emails
to all signups X days before the start of the event (per node
setting), and automatically close signups X hours before their start
(general setting).  There are settings for restricting signups to
selected roles and content types.

There is support for both registered and anonymous users to sign up.
Both can receive confirmation and reminder emails, and registered
users are optionally able to cancel or edit their own signups and view
listings of their current signups.

The Views module (http://drupal.org/project/views) is a dependency
which is used to create various listings, and there is extensive views
support for your own customizations.  Installing the View Bulk
Operations (VBO) module (http://drupal.org/project/views_bulk_operations)
is strongly encouraged to facilitate various administrative tasks.
There is also support to embed the signup form as a pane using panels
(http://drupal.org/project/panels).

Some add-on modules that enhance the basic functionality are available
in the "modules" subdirectory, and there is a listing of other add-on
modules on the Signup project page (http://drupal.org/project/signup).

For installation instructions, see INSTALL.txt.

For instructions on upgrading, see UPGRADE.txt.

For information about theming, see theme/README.txt.

Send feature requests and bug reports to the issue tracking system for
the signup module: http://drupal.org/node/add/project-issue/signup.
For a list of future work, also see http://groups.drupal.org/node/5044.

This module was originally co-developed by Chad Phillips and Jeff
Robbins, and sponsored by Jeff Robbins.  It is now being maintained
by: Derek Wright (http://drupal.org/user/46549) a.k.a. "dww".



CHANGELOG.txt
$Id: CHANGELOG.txt,v 1.126 2011/01/10 02:46:36 dww Exp $

------------------------
6.x-1.x-dev
------------------------
Bugs fixed since 6.x-1.0:
= #856604 by dww: Removed duplicate entry in signup_theme().
= #1020590 by dww: Removed includes/no_views.inc (missed during #582986).
= #1020758 by dww: Fixed missing table prefix for query (introduced in #856604)


------------------------
6.x-1.0 (2011-01-09)
------------------------
Bugs fixed since 6.x-1.0-rc6:
= #597808 by dww: Fixed bug that was causing signup limits to be ignored
  unless the signup_status module is enabled (broken in rc6 by #581652).
= #370397 by wesley_2mpact and ezra-g: All anonymous signups cancelled
  when any user is deleted.
= #676276 by keystr0k and ezra-g: Signup not closed when limit reached
= #605594 by markus_petrux and ezra-g: node_load() during signup's
  node_save() breaks node data cached by CCK
= #719188 by ezra-g: Admins should be able to sign up users regardless
  of their permissions
= #633636 by Magnus: Untranslatable strings in date.inc
= #550036 by iva2k: PHP warnings for node without date value
= #865108 by dooug: Minor typo - Views integration signup.theme ->
  views/signup_form.inc
= #858592 by maker, greggles: Signup Administration Page Error with Postgres

Other changes since 6.x-1.0-rc6:
= #582986 by dww and ezra-g: Made views a required dependency and
  ditched non-views code
= #658214 by Dave Reid: Added vertical tabs support.
= #856604 by ezra-g, jenlampton, jm.federico, coltrane: Made event
  signups visible on user registration form
= #226204 by dww: Exposed the numeric signup ID (unique site-wide) as a token.
= #656250 by Peter Bex, maverick14: Add SID field Views and Views
  token integration
= #594686 by dww: Split all default views into their own separate files.
= #592202 by dww: Added a pager and increased items per page on default views.
= #696010 by ezra-g: Alterable signup confirmation drupal_set_message()
= #856810 by ezra-g: Option to return array from theme_signup_user_form()
= #670944</a> by mlsamuelson: Provided a means to reset node_load()'s
  static node caching when calling signup_sign_up_user()
= #341920 by dww, Eirik_R and ezra-g: Removed work-arounds for DateAPI
  limitations now taht new official releases of DateAPI are out


------------------------
6.x-1.0-rc6 (2009-09-23)
------------------------
Bugs fixed since 6.x-1.0-rc5:
= #439294 by dww: Fixed bug where %user_signup_info was not being
  replaced with an empty string on sites that do not define any extra
  fields on the signup form.
= #539052 by dww: Fixed jQuery bug when users edited their signups that
  was causing most of the form elements to be enabled before pressing
  the 'Edit' button (all form elements should be disabled at first).
= #582626 by dww: Fixed bug in the visibility logic for the "Signup:
  Node: Signup link" views field handler which used to think that if
  signups were closed on a node that the node wasn't signup-enabled.
  This would cause the Signup link fields to only appear on nodes where
  signups were open.
= #578278 by neil.brown, dww: Fixed bug where the signup_confirm_email
  field was shown to administrators editing another user's signup.

API and schema changes since 6.x-1.0-rc5:
= #581734 by dww: Improved the Signup API by adding some much needed hooks:
  - hook_signup_data_alter() to modify a $signup object before it's saved
  - hook_signup_insert() invoked when a signup is inserted into the DB
  - hook_signup_update() invoked when a signup is updated in the DB
= #581652 by dww: Added a {signup_log}.count_towards_limit column and
  adjusted the limit logic to only operate on the effective total (the
  SUM() of {signup_log}.count_towards_limit) not the total # of signups.
  Both signup_total and signup_effective_total are loaded into $node.

Other changes since 6.x-1.0-rc5:
= #578502 by dww: Moved signup-related settings from node/N/edit to a
  new "Settings" subtab at node/N/signups/settings.
= #578592 by dww: Cleaned up and fixed the "Signup summary" at the top
  of the node/N/signups/admin. The UI also makes sense now for signups
  that don't count towards the signup limit (#581652).
= #578344 by dww: Added a "Signup" group on the module administration
  page (admin/build/modules) for all the signup-related modules.


------------------------
6.x-1.0-rc5 (2009-08-23)
------------------------
Bugs fixed since 6.x-1.0-rc4:
= #533970 by dww: Fixed bug that caused the "signups" tab on nodes to
  generate an "Access denied" message unless the signup attendee list
  was configured to appear as a tab (introduced in RC4 via #529478).
= #534948 by dww: Fixed bugs where signup_confirm_email.module breaks
  signup_status.module due to the unorthodox way it was trying to
  hijack the signup form for the confirmation.  Now, confirmation is
  handled via a checkbox (which is only shown if needed via JavaScript)
  instead of trying to redirect to a separate confirmation form.
= #536094 by dww: Fixed a UI bug when editing existing signups. If
  there was a form validation error (e.g. a required field was left
  blank), the form now reloads already in "edit" mode with all the
  fields enabled and the submit button already converted to "Save."
= #544882 by dereine: Fixed a few places that called node_load() so that
  they can now make use of the node caching feature.

New features since 6.x-1.0-rc4:
= #549646 by dww: Exposed signup-related node and signup data to token.module.


------------------------
6.x-1.0-rc4 (2009-07-27)
------------------------
Bugs fixed since 6.x-1.0-rc3:
= #370397 by ezra-g: Fixed critical data-loss bug where all anonymous
  signups would be destroyed if any authenticated user was deleted.
  This bug was introduced during the port to Drupal 6 core, due to an
  undocumented API change between Drupal core versions 5 and 6.
= #350548 by dww: Fixed logic bug in the support for Event module which

signup.api.php
<?php
// $Id: signup.api.php,v 1.5 2009/09/19 01:42:52 dww Exp $


/**
 * @file
 * This file documents the hooks invoked by the Signup module.
 */

/**
 * Hook to alter signup data before a signup is inserted or updated.
 *
 * @param $signup
 *   Reference to the fully-loaded signup object representing the signup.
 * @param $form_values
 *   Array of form values (if any) from the signup being inserted or updated.
 */
function hook_signup_data_alter(&$signup, $form_values) {
  // TODO
}

/**
 * Hook invoked when a signup is being canceled.
 *
 * At the time this hook is invoked the record about the signup in the
 * {signup_log} table still exists, but the node has already had its signup
 * total decremented.
 *
 * @param $node
 *   The fully-loaded node object that the signup is being canceled from.
 * @param $signup
 *   An object containing all the known information about the signup being
 *   canceled. Contains all the data from the {signup_log} row representing
 *   the canceled signup. See the schema definition for descriptions of each
 *   field and what they represent.
 *
 * @return
 *   Ignored.
 *
 * @see signup_cancel_signup()
 */
function hook_signup_cancel($signup, $node) {
  $info = array();
  $info[] = t('Signup ID: @sid', array('@sid' => $signup->sid));
  $info[] = t('Node ID: @nid', array('@nid' => $signup->nid));
  $info[] = t('User ID: @uid', array('@uid' => $signup->uid));
  $info[] = t('Email address for anonymous signup: @anon_mail', array('@anon_mail' => $signup->anon_mail));
  $info[] = t('Date/time when the signup was created: @signup_time', array('@signup_time' => $signup->signup_time));
  $form_data = unserialize($signup->form_data);
  $info[] = t('Custom signup form data: %signup_form_data', array('%signup_form_data' => theme('signup_custom_data_email', $form_data)));
  $info[] = t('Attendance record: %attended', array('%attended' => theme('signup_attended_text', $signup->attended)));
  $info[] = t('Slots consumed by this signup: @count_towards_limit', array('@count_towards_limit' => $signup->count_towards_limit));

  drupal_set_message(theme('item_list', $info, t('Signup canceled for %node_title', array('%node_title' => $node->title))));
}

/**
 * Hook invoked after a signup has been inserted.
 *
 * @param $signup
 *   The fully-loaded signup object representing the new signup.
 */
function hook_signup_insert($signup) {
  // TODO
}

/**
 * Hook invoked after a signup has been updated.
 *
 * @param $signup
 *   The fully-loaded signup object representing the updated signup.
 */
function hook_signup_update($signup) {
  // TODO
}

/**
 * Hook invoked when a signup is being created to gather other signup data.
 *
 * This hook allows other modules to inject information into the custom signup
 * data for each signup.  The array is merged with the values of any custom
 * fields from theme_signup_user_form(), serialized, and stored in the
 * {signup_log} database table.
 *
 * @param $node
 *   Fully-loaded node object being signed up to.
 * @param $account
 *   Full-loaded user object who is signing up.
 *
 * @return
 *   Keyed array of fields to include in the custom data for this signup. The
 *   keys for the array are used as labels when displaying the field, so they
 *   should be human-readable (and wrapped in t() to allow translation).
 *
 * @see signup_sign_up_user()
 * @see theme_signup_user_form()
 */
function hook_signup_sign_up($node, $account) {
  return array(
    t('Node type') => node_type_get_name($node->type),
    t('User created') => format_date($account->created),
  );
}


/**
 * Hook invoked whenever a node is reopened for signups.
 *
 * A node with signups closed could be reopened in two main cases: 1) someone
 * cancels a signup and the signup limit is no longer reached; 2) a signup
 * administrator manually re-opens signups.
 *
 * @param $node
 *   Fully-loaded node object that is now open for signups.
 *
 * @return
 *   Ignored.
 *
 * @see signup_open_signup()
 */
function hook_signup_open($node) {
  drupal_set_message(t('Duplicate message: signups are now open on %title.', array('%title' => $node->title)));
}


/**
 * Hook invoked whenever a node is closed for signups.
 *
 * Signups are closed in 3 main cases: 1) it is a time-based node and the
 * close-in-advance time has been reached (auto-close via cron); 2) the node
 * has a signup limit and the limit is reached; 3) a signup administrator
 * manually closes signups.
 *
 * @param $node
 *   Fully-loaded node object that is now closed for signups.
 *
 * @return
 *   Ignored.
 *
 * @see signup_close_signup()
 */
function hook_signup_close($node) {
  drupal_set_message(t('Duplicate message: signups are now closed on %title.', array('%title' => $node->title)));
}


/**
 * Hook invoked to see if signup information should be printed for a node.
 *
 * This hook is invoked whenever someone is viewing a signup-enabled node and
 * allows modules to suppress any signup-related output.  If any module's
 * implementation of this hook returns TRUE, no signup information will be
 * printed for that node.
 *
 * @param $node
 *   The fully-loaded node object being viewed.
 *
 * @return
 *   TRUE if you want to prevent signup information from being printed, FALSE
 *   or NULL if the information should be printed.
 *
 * @see _signup_needs_output()
 * @see _signup_menu_access()
 * @see signup_nodeapi()
 */
function hook_signup_suppress($node) {
  if ($node->nid % 2) {
    drupal_set_message(t('Signup information suppressed for odd node ID %nid.', array('%nid' => $node->nid)));
    return TRUE;
  }
}

/**
 * Hook invoked to control access to signup menu items.
 *
 * This hook is invoked to check access for signup menu items, in particular,
 * the signup-related tabs on signup-enabled nodes. If no value is returned
 * (NULL), the hook is ignored and the usual access logic is enforced via the
 * Signup module. If multiple modules return a value, the logical OR is used,
 * so if anyone returns TRUE, access is granted. If everyone returns FALSE,
 * access is denied (even if the Signup module would normally grant access).
 *
 * @param $node
 *   The fully-loaded node object where the menu items would be attached.
 * @param $menu_type
 *   String specifying what kind of menu item to test access for. Can be:
 *   'signup': the signup form
 *   'list': the signup attendee listing
 *   'admin': the signup administration tab
 *   'add': the signup administration tab to add other users (requires
 *          that signups are currently open on the given node).
 *   'broadcast': for the broadcast tab
 *
 * @return
 *   TRUE if you want to allow access to the requested menu item, FALSE if you
 *   want to deny access (although if another hook implementation returns
 *   TRUE, that will take precedence), or NULL if you don't care and want to
 *   let Signup module itself decide access based on its own logic.
 *
 * @see _signup_menu_access()
 */
function hook_signup_menu_access($node, $menu_type) {
  // For example, you might want to test that the current user is the
  // administrator of the organic group that a signup-enabled node belongs to,
  // in which case, you'd return TRUE here to give that user full signup
  // powers over events in their group.
}



js/admin.settings.js
/* $Id: admin.settings.js,v 1.6 2009/01/24 08:52:40 dww Exp $ */
(function ($) {

/**
 * Conditionally show/hide settings based on the signup form location.
 *
 * If the signup form is being show on the node itself, it's wrapped
 * in a fieldset and there's a setting to control if that fieldset
 * should be collapsed.  Only show this setting if the form is going
 * to be displayed on the node itself.
 */
Drupal.behaviors.signupShowFormLocationSetting = {
  attach: function (context, settings) {
    $('div.signup-form-location-radios input.form-radio', context).click(function () {
      // Simple part: Depending on the form location, hide/show the
      // collapsible fieldset setting.
      if (this.value == 'node') {
        $('div.signup-fieldset_collapsed-setting', context).show();
      }
      else {
        $('div.signup-fieldset_collapsed-setting', context).hide();
      }
    });
  }
};

/**
 * Conditionally show/hide settings based on the signup user list setting.
 *
 * If the signup user list is going to be an embedded view, show the
 * view-related settings, otherwise, hide them.
 */
Drupal.behaviors.signupShowUserListViewSetting = {
  attach: function (context, settings) {
    $('div.signup-display-signup-user-list-setting input.form-radio', context).click(function () {
      if (this.value == 'embed-view' || this.value == 'embed-view-tab') {
        $('div.signup-user-list-view-settings', context).show();
      }
      else {
        $('div.signup-user-list-view-settings', context).hide();
      }
    });
  }
};

/**
 * Conditionally show/hide settings based on the signup admin list setting.
 *
 * If the administer signup user list is going to be an embedded view,
 * show the view-related settings, otherwise, hide them.
 */
Drupal.behaviors.signupShowAdminListViewSetting = {
  attach: function (context, settings) {
    $('div.signup-display-signup-admin-list-setting input.form-radio', context).click(function () {
      if (this.value == 'embed-view') {
        $('div.signup-admin-list-view-settings', context).show();
      }
      else {
        $('div.signup-admin-list-view-settings', context).hide();
      }
    });
  }
};

})(jQuery);


js/signup_edit_form.js
/* $Id: signup_edit_form.js,v 1.4 2009/09/20 22:25:50 dww Exp $ */
(function ($) {

/**
 * On the signup edit form, add some bling to make users "confirm" the edit.
 *
 * When the form first loads, all the form elements are disabled
 * except for the 'Update signup' button.  When a user clicks that,
 * the other form elements are enabled, and the 'Update signup' button
 * text changes into 'Save changes'.
 */
Drupal.behaviors.enableSignupEditForm = {
  attach: function (context, settings) {
    if (!Drupal.settings.signupEditFormErrors) {
      var $button = $('#edit-save', context).click(enableSave);
      var $form = $button.parents('form:first');
      var $form_elements = $form.find(':visible');
      var original_button_title = $button.val();

      $form_elements.attr('disabled', 'disabled');
      $button.attr('disabled', '').val(Drupal.t('Edit'));
    }
   
    function enableSave() {
      $form_elements.attr('disabled', '');
      $button.unbind('click', enableSave).val(original_button_title);
      return false;
    }
  }
};

})(jQuery);


js/admin.content_types.js
/* $Id: admin.content_types.js,v 1.4 2009/04/15 22:25:38 dww Exp $ */
(function ($) {

/**
 * Conditionally show or hide the signup date field setting.
 *
 * On a node type settings form, if the "Allow signups" radios set
 * set to 0 ('Disabled'), or the node type is event-enabled, then
 * hide the date field setting.  Otherwise, show it.
 */
Drupal.behaviors.signupShowDateField = {
  attach: function (context, settings) {
    $('div.signup-node-default-state-radios input[type=radio], div.event-nodeapi-radios input[type=radio]', context).click(function () {
      var eventEnabled = false;
      var signupDisabled = true;
      $('div.event-nodeapi-radios input.form-radio', context).each(function() {
        if (this.checked && this.value != 'never') {
          eventEnabled = true;
        }
      });
      $('div.signup-node-default-state-radios input.form-radio', context).each(function() {
        if (this.checked && this.value != 'disabled') {
          signupDisabled = false;
        }
      });
      if (signupDisabled || eventEnabled) {
        $('div.signup-date-field-setting', context).hide();
      }
      else {
        $('div.signup-date-field-setting', context).show();
      }
    });
  }
};

})(jQuery);


views/default_views/signup_user_list.view.php
<?php
// $Id: signup_user_list.view.php,v 1.3 2011/01/09 18:31:52 dww Exp $

/**
 * @file
 * All users who have signed up for a given node.
 */

$view = new view;
$view->name = 'signup_user_list';
$view->description = 'List of all users signed up for a given node';
$view->tag = 'Signup';
$view->view_php = '';
$view->base_table = 'signup_log';
$view->is_cacheable = FALSE;
$view->api_version = 2;
$view->disabled = FALSE;
$handler = $view->new_display('default', 'Defaults', 'default');
$handler->override_option('relationships', array(
  'uid' => array(
    'label' => 'Signup user',
    'required' => 1,
    'id' => 'uid',
    'table' => 'signup_log',
    'field' => 'uid',
    'relationship' => 'none',
  ),
  'nid' => array(
    'label' => 'Signup node',
    'required' => 1,
    'id' => 'nid',
    'table' => 'signup_log',
    'field' => 'nid',
    'relationship' => 'none',
  ),
));
$handler->override_option('fields', array(
  'name' => array(
    'label' => 'Username',
    'link_to_user' => 1,
    'exclude' => 0,
    'id' => 'name',
    'table' => 'users',
    'field' => 'name',
    'relationship' => 'uid',
  ),
  'signup_time' => array(
    'label' => 'Signup time',
    'date_format' => 'small',
    'custom_date_format' => '',
    'exclude' => 0,
    'id' => 'signup_time',
    'table' => 'signup_log',
    'field' => 'signup_time',
    'relationship' => 'none',
  ),
));
$handler->override_option('arguments', array(
  'nid' => array(
    'default_action' => 'not found',
    'style_plugin' => 'default_summary',
    'style_options' => array(),
    'wildcard' => 'all',
    'wildcard_substitution' => 'All',
    'title' => '',
    'default_argument_type' => 'fixed',
    'default_argument' => '',
    'validate_type' => 'signup_status',
    'validate_fail' => 'not found',
    'break_phrase' => 0,
    'not' => 0,
    'id' => 'nid',
    'table' => 'node',
    'field' => 'nid',
    'relationship' => 'nid',
    'default_options_div_prefix' => '',
    'default_argument_user' => 0,
    'default_argument_fixed' => '',
    'default_argument_php' => '',
    'validate_argument_node_type' => array(),
    'validate_argument_node_access' => 0,
    'validate_argument_nid_type' => 'nid',
    'validate_argument_vocabulary' => array(),
    'validate_argument_type' => 'tid',
    'validate_argument_signup_status' => 'any',
    'validate_argument_signup_node_access' => 1,
    'validate_argument_php' => '',
  ),
));
$handler->override_option('access', array(
  'type' => 'perm',
  'perm' => 'view all signups',
));
$handler->override_option('items_per_page', 100);
$handler->override_option('use_pager', 1);
$handler->override_option('style_plugin', 'table');
$handler->override_option('style_options', array(
  'grouping' => '',
  'override' => 1,
  'sticky' => 0,
  'order' => 'asc',
  'columns' => array(
    'name' => 'name',
    'signup_time' => 'signup_time',
  ),
  'info' => array(
    'name' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'signup_time' => array(
      'sortable' => 1,
      'separator' => '',
    ),
  ),
  'default' => 'signup_time',
));


views/default_views/signup_user_admin_list.view.php
<?php
// $Id: signup_user_admin_list.view.php,v 1.3 2011/01/09 18:31:52 dww Exp $

/**
 * @file
 * Administrative list of all users signed up for a given node.
 */

$view = new view;
$view->name = 'signup_user_admin_list';
$view->description = t('Administrative list of all users signed up for a given node.');
$view->tag = 'Signup';
$view->view_php = '';
$view->base_table = 'signup_log';
$view->is_cacheable = FALSE;
$view->api_version = 2;
$view->disabled = FALSE;
$handler = $view->new_display('default', 'Defaults', 'default');
$handler->override_option('relationships', array(
  'nid' => array(
    'label' => 'Signup node',
    'required' => 1,
    'id' => 'nid',
    'table' => 'signup_log',
    'field' => 'nid',
    'relationship' => 'none',
  ),
  'uid' => array(
    'label' => 'Signup user',
    'required' => 1,
    'id' => 'uid',
    'table' => 'signup_log',
    'field' => 'uid',
    'relationship' => 'none',
  ),
));
$handler->override_option('fields', array(
  'name' => array(
    'label' => 'Username',
    'link_to_user' => 1,
    'exclude' => 0,
    'id' => 'name',
    'table' => 'users',
    'field' => 'name',
    'relationship' => 'uid',
  ),
  'signup_time' => array(
    'label' => 'Signup time',
    'date_format' => 'small',
    'custom_date_format' => '',
    'exclude' => 0,
    'id' => 'signup_time',
    'table' => 'signup_log',
    'field' => 'signup_time',
    'relationship' => 'none',
  ),
  'form_data' => array(
    'label' => 'Additional info',
    'type' => 'ul',
    'separator' => ', ',
    'empty' => '',
    'exclude' => 0,
    'id' => 'form_data',
    'table' => 'signup_log',
    'field' => 'form_data',
    'relationship' => 'none',
    'form_data_fieldname' => '',
  ),
  'attended' => array(
    'label' => 'Attendance',
    'exclude' => 0,
    'id' => 'attended',
    'table' => 'signup_log',
    'field' => 'attended',
    'relationship' => 'none',
  ),
  'edit_link' => array(
    'label' => 'Operations',
    'text' => '',
    'exclude' => 0,
    'id' => 'edit_link',
    'table' => 'signup_log',
    'field' => 'edit_link',
    'relationship' => 'none',
  ),
));
$handler->override_option('arguments', array(
  'nid' => array(
    'default_action' => 'ignore',
    'style_plugin' => 'default_summary',
    'style_options' => array(),
    'wildcard' => 'all',
    'wildcard_substitution' => 'All',
    'title' => '',
    'default_argument_type' => 'fixed',
    'default_argument' => '',
    'validate_type' => 'signup_status',
    'validate_fail' => 'not found',
    'break_phrase' => 0,
    'not' => 0,
    'id' => 'nid',
    'table' => 'node',
    'field' => 'nid',
    'relationship' => 'nid',
    'default_options_div_prefix' => '',
    'default_argument_user' => 0,
    'default_argument_fixed' => '',
    'default_argument_php' => '',
    'validate_argument_node_access' => 0,
    'validate_argument_nid_type' => 'nid',
    'validate_argument_vocabulary' => array(),
    'validate_argument_type' => 'tid',
    'validate_argument_signup_status' => 'any',
    'validate_argument_signup_node_access' => 1,
    'validate_argument_php' => '',
  ),
));
$handler->override_option('access', array(
  'type' => 'none',
));
$handler->override_option('items_per_page', 100);
$handler->override_option('use_pager', 1);
$handler->override_option('style_plugin', 'table');
$handler->override_option('style_options', array(
  'grouping' => '',
  'override' => 1,
  'sticky' => 0,
  'order' => 'desc',
  'columns' => array(
    'name' => 'name',
    'signup_time' => 'signup_time',
    'form_data' => 'form_data',
    'attended' => 'attended',
    'edit_link' => 'edit_link',
  ),
  'info' => array(
    'name' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'signup_time' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'form_data' => array(
      'separator' => '',
    ),
    'attended' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'edit_link' => array(
      'separator' => '',
    ),
  ),
  'default' => 'signup_time',
));


views/default_views/signup_user_vbo_admin_list.view.php
<?php
// $Id: signup_user_vbo_admin_list.view.php,v 1.3 2011/01/09 18:31:52 dww Exp $

/**
 * @file
 * Views Bulk Operations (VBO) view for administering signups on a node.
 */

$view = new view;
$view->name = 'signup_user_vbo_admin_list';
$view->description = t('Per-node signup administration interface with checkboxes for bulk operations.');
$view->tag = 'Signup';
$view->view_php = '';
$view->base_table = 'signup_log';
$view->is_cacheable = FALSE;
$view->api_version = 2;
$view->disabled = FALSE;
$handler = $view->new_display('default', 'Defaults', 'default');
$handler->override_option('relationships', array(
  'nid' => array(
    'label' => 'Signup node',
    'required' => 1,
    'id' => 'nid',
    'table' => 'signup_log',
    'field' => 'nid',
    'relationship' => 'none',
  ),
  'uid' => array(
    'label' => 'Signup user',
    'required' => 1,
    'id' => 'uid',
    'table' => 'signup_log',
    'field' => 'uid',
    'relationship' => 'none',
  ),
));
$handler->override_option('fields', array(
  'name' => array(
    'label' => 'Username',
    'link_to_user' => 1,
    'exclude' => 0,
    'id' => 'name',
    'table' => 'users',
    'field' => 'name',
    'relationship' => 'uid',
  ),
  'signup_time' => array(
    'label' => 'Signup time',
    'date_format' => 'small',
    'custom_date_format' => '',
    'exclude' => 0,
    'id' => 'signup_time',
    'table' => 'signup_log',
    'field' => 'signup_time',
    'relationship' => 'none',
  ),
  'form_data' => array(
    'label' => 'Additional info',
    'type' => 'ul',
    'separator' => ', ',
    'empty' => '',
    'exclude' => 0,
    'id' => 'form_data',
    'table' => 'signup_log',
    'field' => 'form_data',
    'relationship' => 'none',
    'form_data_fieldname' => '',
  ),
  'attended' => array(
    'label' => 'Attendance',
    'exclude' => 0,
    'id' => 'attended',
    'table' => 'signup_log',
    'field' => 'attended',
    'relationship' => 'none',
  ),
  'edit_link' => array(
    'label' => 'Operations',
    'text' => '',
    'exclude' => 0,
    'id' => 'edit_link',
    'table' => 'signup_log',
    'field' => 'edit_link',
    'relationship' => 'none',
  ),
));
$handler->override_option('arguments', array(
  'nid' => array(
    'default_action' => 'ignore',
    'style_plugin' => 'default_summary',
    'style_options' => array(),
    'wildcard' => 'all',
    'wildcard_substitution' => 'All',
    'title' => '',
    'default_argument_type' => 'fixed',
    'default_argument' => '',
    'validate_type' => 'signup_status',
    'validate_fail' => 'not found',
    'break_phrase' => 0,
    'not' => 0,
    'id' => 'nid',
    'table' => 'node',
    'field' => 'nid',
    'relationship' => 'nid',
    'default_options_div_prefix' => '',
    'default_argument_user' => 0,
    'default_argument_fixed' => '',
    'default_argument_php' => '',
    'validate_argument_node_access' => 0,
    'validate_argument_nid_type' => 'nid',
    'validate_argument_vocabulary' => array(),
    'validate_argument_type' => 'tid',
    'validate_argument_signup_status' => 'any',
    'validate_argument_signup_node_access' => 1,
    'validate_argument_php' => '',
  ),
));
$handler->override_option('access', array(
  'type' => 'none',
));
$handler->override_option('items_per_page', 100);
$handler->override_option('use_pager', 1);
$handler->override_option('style_plugin', 'bulk');
$handler->override_option('style_options', array(
  'grouping' => '',
  'override' => 1,
  'sticky' => 0,
  'order' => 'asc',
  'columns' => array(
    'name' => 'name',
    'signup_time' => 'signup_time',
    'form_data' => 'form_data',
    'attended' => 'attended',
    'edit_link' => 'edit_link',
  ),
  'info' => array(
    'name' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'signup_time' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'form_data' => array(
      'separator' => '',
    ),
    'attended' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'edit_link' => array(
      'separator' => '',
    ),
  ),
  'default' => '-1',
  'execution_type' => '1',
  'display_type' => '0',
  'skip_confirmation' => 0,
  'display_result' => 1,
  'selected_operations' => array(
    '5c7373a6aa7a8bf5549412c9f887c2d3' => '5c7373a6aa7a8bf5549412c9f887c2d3',
    'd3a1a31a80ae6f3a3445aa18d827c62e' => 'd3a1a31a80ae6f3a3445aa18d827c62e',
    '486e69f274fc0c7fb96d00a55ce752ae' => '486e69f274fc0c7fb96d00a55ce752ae',
  ),
));
